src/assets/styles/styles.11ty.js

```diff
@@ -0,0 +1,188 @@
+https://github.com/fluid-project/fluidic-11ty/raw/master/AUTHORS.md.
+
+Licensed under the New BSD license. You may not use this file except in compliance with this License.
+
+You may obtain a copy of the New BSD License at
+https://github.com/fluid-project/fluidic-11ty/raw/master/LICENSE.md.
+*/
+"use strict";
+
+const fs = require("fs");
+const fg = require("fast-glob");
+const path = require("path");
+const sass = require("sass");
+const stylus = require("stylus");
+const autoprefixer = require("autoprefixer");
+const postcss = require("postcss");
+const CleanCSS = require("clean-css");
+const cssesc = require("cssesc");
+
+const isProd = process.env.ELEVENTY_ENV === "production";
+
+/**
+ * @see https://github.com/maxboeck/eleventastic/blob/master/src/assets/styles/styles.11ty.js
+ */
+
+module.exports = class {
+    // Configure the transform.
+    async data() {
+        const entryPaths = await fg("./src/assets/styles/style.{css,scss,styl}");
+        const entryPath = entryPaths[0];
+        return {
+            permalink: '/assets/styles/style.css',
+            eleventyExcludeFromCollections: true,
+            entryPath
+        };
+    }
+
+    // Compile SCSS to CSS.
+    async compileScss(options) {
+        return new Promise((resolve, reject) => {
+            const config = {
+                file: options.entryPath,
+                sourceMap: !isProd ? 'out.map' : false,
+                sourceMapEmbed: !isProd || false,
+                outputStyle: isProd ? 'compressed' : 'expanded'
+            };
+
+            sass.render(config, (err, result) => {
+                if (err) {
+                    reject(err);
+                } else {
+                    const processedCss = this.processCss({str: result.css, entryPath: options.entryPath});
+                    resolve(processedCss);
+                }
+            });
+        });
+    }
+
+    // Compile Stylus to CSS.
+    async compileStylus(options) {
+        return new Promise((resolve, reject) => {
+            const sourceMap = isProd ? false : { inline: true };
+
+            stylus(options.str)
+                .set('filename', options.entryPath)
+                .set('sourcemap', sourceMap)
+                .set('compress', isProd)
+                .render((err, css) => {
+                    if (err) {
+                        reject(err);
+                    } else {
+                      const processedCss = this.processCss({str: css, entryPath: options.entryPath});
+                      resolve(processedCss);
+                    }
+                });
+        });
+    }
+
+    // Process CSS with PostCSS.
+    async processCss(options) {
+      const from = options.entryPath || 'undefined';
+      const css = postcss([autoprefixer]).process(options.str, {from}).css;
+      return css;
+    }
+
+    // Compile and process CSS (source-agnostic).
+    async compile(options) {
+      let css;
+      switch (options.fileType) {
+        case 'css':
```

Comment:
Instead of using a switch statement it might be better to just map the file type to functions. 

```javascript
return processor[options.type](options);
```

Reply:
@jobara would the idea be to create an object, `processor`, with the method names set as key values like this?:

```javascript
const processor = {
    css: this.processCss,
    scss: this.compileScss,
    styl: this.compileStylus
};
```

When I tried this and called the method like so:

```javascript
return processor[options.fileType](options);
```

I got a puzzling error when trying to process Sass or Stylus:

```
Uncaught exception: (more in DEBUG output)
> NoSuchMethodError: method not found: 'processCss' (this.processCss is not a function)

`` was thrown:
    TypeError: this.processCss is not a function
        at /Users/ned/Sites/fluidic-11ty/src/assets/styles/styles.11ty.js:58:47
```
That line reference points here: https://github.com/fluid-project/fluidic-11ty/pull/30/files#diff-91cd3b573ebc02e9435452bb65058a3eR58 Any ideas?

Reply:
@greatislander link to the lines that are being replaced doesn't seem to work anymore. However, I think you probably did the following:

```javascript
    // Compile and process CSS (source-agnostic).
    async compile(options) {
        const processor = {
            css: this.processCss,
            scss: this.compileScss,
            styl: this.compileStylus
        };

        return processor[options.fileType](options);
    }
```

I believe what's happening is that the `this` reference is actually the `processor` object and not the module. 

Do you need to export the module as a `class` or can you just export an Object or Function? You may want to just create a top level object that you export and use that as a namespace for all the functions. For example would something like the following work?

```javascript

const styles = {};

styles.data = async function () {...};

styles.processor.css = async function () {...};
styles.processor.scss = async function () {...};
styles.processor.stylus = async function () {...};

...

styles.compile = async function (options) {
    return styles.processor[options.fileType](options);
}

...

module.exports = styles;

```

Reply:
It has to be a class as that's how the 11ty.js templating system works: https://www.11ty.dev/docs/languages/javascript/#classes